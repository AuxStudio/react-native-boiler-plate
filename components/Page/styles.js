import { StyleSheet, Platform } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'stretch',
    backgroundColor: styleConstants.colors.white,
    paddingTop: Platform.OS === 'ios' ? 22 : 0,
  },
});

export default styles;
