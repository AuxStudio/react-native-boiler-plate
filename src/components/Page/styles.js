import { StyleSheet, Platform } from 'react-native';

import { colors } from '../../static/styleConstants';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'stretch',
    backgroundColor: colors.white,
    paddingTop: Platform.OS === 'ios' ? 22 : 0,
  },
});

export default styles;
