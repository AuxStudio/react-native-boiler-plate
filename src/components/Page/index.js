import React from 'react';
import PropTypes from 'prop-types';
import { View, ViewPropTypes } from 'react-native';
import { StatusBar } from 'react-native-simple-components';

import styleConstants from '../../styleConstants';

import styles from './styles';

const propTypes = {
  verticalCenter: PropTypes.bool,
  horizontalCenter: PropTypes.bool,
  children: PropTypes.node,
  style: ViewPropTypes.style,
  testID: PropTypes.string,
};

const defaultProps = {};

const Page = ({ verticalCenter, horizontalCenter, children, style, testID }) => {
  const verticalCenterStyles = verticalCenter && {
    justifyContent: 'center',
  };
  const horizontalCenterStyles = horizontalCenter && {
    alignItems: 'center',
  };

  return (
    <View
      style={[styles.container, verticalCenterStyles, horizontalCenterStyles, style]}
      testID={testID}
    >
      <StatusBar backgroundColor={styleConstants.colors.primary} barStyle="light-content" />

      {children}
    </View>
  );
};

Page.propTypes = propTypes;
Page.defaultProps = defaultProps;

export default Page;
