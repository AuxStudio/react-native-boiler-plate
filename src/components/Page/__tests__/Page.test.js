import React from 'react';
import renderer from 'react-test-renderer';

import Page from '..';

describe('Page', () => {
  describe('renders', () => {
    it('renders', () => {
      const component = renderer.create(<Page />);

      expect(component).toMatchSnapshot();
    });

    it('renders the verticalCenter styles', () => {
      const component = renderer.create(<Page verticalCenter />);

      expect(component).toMatchSnapshot();
    });

    it('renders the horizontalCenter styles', () => {
      const component = renderer.create(<Page horizontalCenter />);

      expect(component).toMatchSnapshot();
    });
  });
});
