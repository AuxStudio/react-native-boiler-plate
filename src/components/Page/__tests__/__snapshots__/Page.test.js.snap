// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingTop": 22,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#3D84A8",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Page renders renders the horizontalCenter styles 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingTop": 22,
      },
      undefined,
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#3D84A8",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Page renders renders the verticalCenter styles 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingTop": 22,
      },
      Object {
        "justifyContent": "center",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#3D84A8",
          },
        ]
      }
    />
  </View>
</View>
`;
